console-subscriber/src/aggregator.rs

```diff
@@ -85,28 +141,100 @@ struct Stats {
     last_wake: Option<SystemTime>,
 
     poll_times_histogram: Histogram<u64>,
+    poll_stats: PollStats,
+}
+
+struct AsyncOp {
+    id: span::Id,
+    metadata: &'static Metadata<'static>,
+    source: String,
 }
 
 #[derive(Default)]
-struct TaskData<T> {
-    data: HashMap<span::Id, (T, bool)>,
+struct AsyncOpStats {
+    created_at: Option<SystemTime>,
+    closed_at: Option<SystemTime>,
+    latest_poll_op: Option<proto::MetaId>,
+    resource_id: Option<span::Id>,
+    task_id: Option<span::Id>,
+    poll_stats: PollStats,
 }
 
-struct Task {
+struct ResourceOp {
+    id: proto::MetaId,
     metadata: &'static Metadata<'static>,
-    fields: Vec<proto::Field>,
+    resource_id: span::Id,
+    op_name: String,
+    op_type: OpType,
+}
+
+struct IdData<T> {
+    data: HashMap<span::Id, (T, bool)>,
+}
+
+impl Closable for ResourceStats {
+    fn closed_at(&self) -> &Option<SystemTime> {
+        &self.closed_at
+    }
+}
+
+impl Closable for TaskStats {
+    fn closed_at(&self) -> &Option<SystemTime> {
+        &self.closed_at
+    }
+}
+
+impl Closable for AsyncOpStats {
+    fn closed_at(&self) -> &Option<SystemTime> {
+        &self.closed_at
+    }
+}
+
+impl PollStats {
+    fn update_on_span_enter(&mut self, timestamp: SystemTime) {
+        if self.current_polls == 0 {
+            self.last_poll_started = Some(timestamp);
+            if self.first_poll == None {
+                self.first_poll = Some(timestamp);
+            }
+            self.polls += 1;
+        }
+        self.current_polls += 1;
+    }
+    fn update_on_span_exit(&mut self, timestamp: SystemTime) {
```

Comment:
nit: prefer whitespace between function definitions
```suggestion
    }
    
    fn update_on_span_exit(&mut self, timestamp: SystemTime) {
```